### DISCUSSION TITLE
RabbitMQ stops processing any messages when there are unhandled RPC messages

### DISCUSSION THREAD

**Discussion by car1ot** [rabbitmq/rabbitmq-server#7180] [Other]
There are microservices architecture. The microservices communicate with each other using RabbitMQ RPC direct messages. However, if one of the microservices crashes or there is another reason for the Rabbit consumer to shut down, messages may become stuck in the queue. In this scenario, approximately 1000 messages are waiting to be delivered.

When there are unhandled messages in the queue, RabbitMQ stops delivering any messages and becomes really stuck. This means that no new messages are being processed! Even if they are not related to the unhandled messages!

**Comment by michaelklishin** [comment]
Consider using Discussions for questions, they have been around for more than a year. Our team uses issues for specific, actionable, reasonably well understood and researched tasks.

We cannot suggest anything without an executable way to reproduce or at least server logs. For example, you consumers may be [using manual acknowledgements with a prefetch setting of 1](https://rabbitmq.com/confirms.html#channel-qos-prefetch).

Start with [tutorial 2](https://rabbitmq.com/getstarted.html).
